activerecord/lib/active_record/connection_adapters/abstract/quoting.rb

```diff
@@ -17,7 +22,11 @@ def quote(value)
       # SQLite does not understand dates, so this method will convert a Date
       # to a String.
       def type_cast(value, column = nil)
+        value = id_value_for_database(value) if value.is_a?(Base)
```

Comment:
If there is a potential bug in the current code we should add a test to it and fix before changing the behavior. Care to add the test?